schema/in_memory.go

```diff
@@ -0,0 +1,72 @@
+// Copyright © 2024 Meroxa, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package schema
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"sync"
+
+	"github.com/conduitio/conduit-commons/schema"
+	"github.com/conduitio/conduit-connector-protocol/pconduit"
+)
+
+var ErrSchemaNotFound = errors.New("schema not found")
```

Comment:
Errors, that's another thing we need to handle end-to-end! This specific error will only be returned by the in-memory service and not when the plugin actually runs in Conduit. In other words, if a test is asserting that this error is returned, it will test a functionality that actually doesn't exist.

So I think we have to go back to the connector protocol and define the errors that the schema service can return and make sure we create them when unwrapping the gRPC error. Then we can create global variables here that simply point to the errors we define in the protocol, so it's easier for connector developers to use.

Reply:
I added the errors I identified in this PR https://github.com/ConduitIO/conduit-connector-protocol/pull/108. The idea is to create mappings here in this package and Conduit will make sure to return these errors.

```suggestion
var (
	ErrSchemaNotFound       = pconduit.ErrSchemaNotFound
	ErrInvalidSchemaSubject = pconduit.ErrInvalidSchemaSubject
	ErrInvalidSchemaType    = pconduit.ErrInvalidSchemaType
	ErrInvalidSchemaBytes   = pconduit.ErrInvalidSchemaBytes
)
```